src/animation/controller.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2022 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{AnimationStatus, AnimationDirection};
+
+use crate::RequestCtx;
+
+/// Control animations. An Animation controller produces values between 0.0 and 1.0 during
+/// the given duration. You can run this animation forward, backwards,
+/// altered and repeated.
+pub struct AnimationController {
+    duration: f64,
```

Comment:
Maybe better to use `std::time::Duration` as it's more descriptive

Reply:
Duration is defined as:

```
pub struct Duration {
    secs: u64,
    nanos: u32, // Always 0 <= nanos < NANOS_PER_SEC
}
```
If you use that, you constantly convert from integer to float without any real gain.